2_process/src/match_sites_reaches.R

```diff
@@ -27,36 +30,53 @@ get_site_flowlines <- function(reach_sf, sites, sites_crs, max_matches = 1, sear
   reaches_nhd_fields <- reach_sf %>%
     rename(COMID = subsegid) %>%
     mutate(REACHCODE = COMID, ToMeas = 100, FromMeas = 100) %>%
-    st_as_sf()
+    # project reaches to Albers Equal Area Conic so that offsets returned by
+    # get_flowline_index are in meters rather than degrees
+    st_transform(5070)
   
   sites_sf <- sites %>% rowwise() %>%
     filter(across(c(lon, lat), ~ !is.na(.x))) %>%
     mutate(Shape = list(st_point(c(lon, lat), dim = "XY"))) %>%
     st_as_sf() %>% st_set_crs(sites_crs) %>%
     st_transform(st_crs(reaches_nhd_fields)) %>%
     st_geometry()
+  
+  # Indicate the site numbers (id's) to retain based on retain_sites
+  retain_rows <- which(sites$site_id %in% retain_sites)
+  
   message('matching flowlines with reaches...')
+  # Below, precision indicates the resolution of measure precision (in meters) in the output;
+  # since we are interested in a more accurate estimate of the `offset` distance between a 
+  # point and the matched reach, set precision to 1 m.
+  # Conduct initial search using a larger radius (search_radius*2) than specified to 
+  # account for any uncertainty in the RANN:nn2 radius search. Then 
+  # filter sites to include those within the specified search_radius.
   flowline_indices <- nhdplusTools::get_flowline_index(flines = reaches_nhd_fields,
                                                        points = sites_sf,
                                                        max_matches = max_matches,
-                                                       search_radius = search_radius) %>%
+                                                       search_radius = search_radius*2,
```

Comment:
Why *2 now? Is this to allow NWIS sites to be matched and later retained? If that's why, I think it would be better to add a separate search radius column for the NWIS sites to retain. This would allow for still retaiing NWIS sites when the `search_radius` is reduced to something smaller than would match the NWIS sites

Reply:
Yeah, good question. I made this change to initially use a wider search radius when we apply `nhdplusTools::get_flowline_index()`. That function uses a radius search implemented in `RANN::nn2()` and I noticed when looking at DO data that a couple sites were not being picked up with a 500 m search radius (but were picked up with a 1000 m radius) even though the distance between those sites was < 500 m, as confirmed by the offset returned and separately using {sf}. More details are [here](https://github.com/USGS-R/drb-do-ml/pull/67#discussion_r827244387). 

So this doesn't have anything to do with the two NWIS sites but rather, to ensure we're picking up all the sites we would expect initially, which we then filter to the requested `search_radius` a couple lines down. I'm open to other suggestions, too, so happy to hear your thoughts!

Reply:
Strange that those 2 sites do not get picked up! I'm not familiar with the `nn2` function, and the source is in C, so I'm less inclined to look into if it's a matter of using a different argument in `get_flowline_index`

Reply:
Yeah, I wasn't worried about those two sites in particular, but the behavior was puzzling to me. The distances returned seem to be correct, so for now I decided to just adjust the argument in `get_flowline_index` to make sure we're capturing all the sites we'd expect. I tried to explain this reasoning in the comments because I know the different argument to search_radius is a little odd.